frontends/api/src/hooks/learningResources.ts

```diff
@@ -0,0 +1,31 @@
+import { useQuery } from "@tanstack/react-query"
+import { learningResourcesApi as api } from "../clients"
+import type {
+  LearningResourcesApiLearningResourcesListRequest,
+  LearningResourcesApiLearningResourcesRetrieveRequest
+} from "../generated"
+import { createQueryKeys } from "@lukemorales/query-key-factory"
+
+const learningResources = createQueryKeys("learningResources", {
+  detail: (params: LearningResourcesApiLearningResourcesRetrieveRequest) => ({
+    queryKey: [params],
+    queryFn:  () => api.learningResourcesRetrieve(params).then(res => res.data)
+  }),
+  list: (params: LearningResourcesApiLearningResourcesListRequest) => ({
+    queryKey: [params],
+    queryFn:  () => api.learningResourcesList(params).then(res => res.data)
+  })
+})
```

Comment:
`@lukemorales/query-key-factory` is a (very small) library for managing query keys with react query. It's very similar to what we were doing in https://github.com/mitodl/mit-open/blob/e867e5e9fa0e2325836f7d757eb4a27814587a9e/frontends/infinite-corridor/src/api/learning-resources/urls.ts#L209 but a bit less manual / more organized.

See also:
- https://github.com/lukemorales/query-key-factory
- https://tanstack.com/query/v4/docs/react/community/lukemorales-query-key-factory